src/main/java/core/basesyntax/service/DataConverterImpl.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+
+    public static final int TYPE = 0;
+    public static final int FRUIT = 1;
+    public static final int QUANTITY = 2;
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : inputReport) {
+            String[] words = line.split(",");
+            String type = words[TYPE];
+            String fruit = words[FRUIT];
+            String quantity = words[QUANTITY];
+            switch (type) {
+                case "b" -> transactions.add(new FruitTransaction(FruitTransaction.Operation.BALANCE, fruit,
+                        Integer.parseInt(quantity)));
+                case "s" -> transactions.add(new FruitTransaction(FruitTransaction.Operation.SUPPLY, fruit,
+                        Integer.parseInt(quantity)));
+                case "p" -> transactions.add(new FruitTransaction(FruitTransaction.Operation.PURCHASE, fruit,
+                        Integer.parseInt(quantity)));
+                case "r" -> transactions.add(new FruitTransaction(FruitTransaction.Operation.RETURN, fruit,
+                        Integer.parseInt(quantity)));
+            }
+        }
+        return transactions;
+    }
```

Comment:
The DataConverterImpl class should not use a switch-case construction as per the checklist. Instead, it is recommended to use the Strategy pattern. You can create a map of strategies and pass it to the service that will handle the operations based on the transaction type.